### PR TITLE
Add cache contracts template annotations

### PR DESCRIPTION
**Pull request by jack-worman**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       |
| License       | MIT
| Doc PR        |

Add template annotations so static analysis tool can better understand the code.

### CODE REVIEW

File: src/Symfony/Contracts/Cache/CacheInterface.php [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 use Psr\Cache\CacheItemInterface;
 use Psr\Cache\InvalidArgumentException;
+use Symfony\Component\Cache\CacheItem;
```
Comment (nicolas-grekas) [inline src/Symfony/Contracts/Cache/CacheInterface.php]: referencing the component in the contracts breaks the package principles
the annotation should use `CacheItemInterface|ItemInterface`

Comment (stof) [reply src/Symfony/Contracts/Cache/CacheInterface.php]: should it be a union or an intersection though ?

Comment (nicolas-grekas) [reply src/Symfony/Contracts/Cache/CacheInterface.php]: it's a union: ItemInterface already extends CacheItemInterface, accepting a CacheItemInterface is desired for PSR-6 compat, and ItemInterface for more advanced items.

Comment (jack-worman) [reply src/Symfony/Contracts/Cache/CacheInterface.php]: Done

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @jack-worman.

**Pull request merged**